packages/elements/src/components/ui/buttons/NewButton.tsx

```diff
@@ -0,0 +1,102 @@
+import { IconDefinition } from "@fortawesome/fontawesome-svg-core";
+import { faCheck } from "@fortawesome/free-solid-svg-icons/faCheck";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ButtonProps } from "@stenajs-webui/core";
+import cx from "classnames";
+import * as React from "react";
+import { MouseEventHandler, ReactNode } from "react";
+import { InputSpinner } from "../../..";
+import styles from "./NewButton.module.css";
+
+export type ButtonSize = "normal" | "small" | "large";
+
+export interface NewButtonProps extends ButtonProps {
+  label?: string;
+  loadingLabel?: string;
+  className?: string;
+  size?: ButtonSize;
+  loading?: boolean;
+  success?: boolean;
+  successLabel?: string;
+  disabled?: boolean;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  leftIcon?: IconDefinition;
+  left?: ReactNode;
+  rightIcon?: IconDefinition;
+  right?: ReactNode;
+}
+
+const getButtonLabel = (
+  label: string | undefined,
+  success: boolean,
+  successLabel: string | undefined,
+  loading: boolean,
+  loadingLabel: string | undefined
+): string | null => {
```

Comment:
Let it return undefined, and then `getButtonLabel(a, b, c) || null` in component?